onnxruntime/python/tools/transformers/convert_generation.py

```diff
@@ -407,6 +418,65 @@ def shape_inference(onnx_path: str, use_external_data_format: bool = True):
         logger.warning("Failed to run symbolic shape inference on the model.")
 
 
+def pad_weights_of_logits_matmul(onnx_path: str, use_external_data_format: bool = True) -> bool:
+    """Pad the logits MatMul weight in the provided decoder model, which will be overwritten.
+
+    Args:
+        onnx_path (str): Path of onnx model
+        use_external_data_format(bool): output tensors to external data or not.
+    """
+    decoder_model_proto = onnx.load_model(onnx_path, load_external_data=True)
+
+    logits_output_name = decoder_model_proto.graph.output[0].name
+    
+    decoder_model = OnnxModel(decoder_model_proto)
+
+    output_name_to_node = decoder_model.output_name_to_node()
+    assert logits_output_name in output_name_to_node
+
+    matmul_node = output_name_to_node[logits_output_name]
+    # Sanity check - the logits need to be produced by a MatMul node
+    if matmul_node.op_type != "MatMul":
+        return False
+    
+    # The logits MatMul weight MUST be an initializer
+    logits_weight = decoder_model.get_initializer(matmul_node.input[1])
+    if logits_weight is None:
+        return False
+
+    # The logits MatMul weight MUST be fp16
+    if (logits_weight.data_type != TensorProto.DataType.FLOAT16):
```

Comment:
Could we also handle a case of initializer --> Transpose --> MatMul. It is because the initializer is same as the one in embedding layer. So using Transpose could reduce model size.